src/proto/h2/client.rs

```diff
@@ -164,21 +163,19 @@ where
     // 'Client' has been dropped. This is to get around a bug
     // in h2 where dropping all SendRequests won't notify a
     // parked Connection.
```

Comment:
I definitely don't want to touch this in this PR, but I do wonder if that bug still exists. https://github.com/hyperium/h2/issues/502 seems relevant.

Reply:
Possibly! Would be happy to try having it removed in a separate PR.